libminifi/test/WriteToFlowFileTestProcessor.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "WriteToFlowFileTestProcessor.h"
+
+#include <string>
+#include <vector>
+
+namespace org::apache::nifi::minifi::processors {
+
+const std::string WriteToFlowFileTestProcessor::OnScheduleLogStr = "WriteToFlowFileTestProcessor::onSchedule executed";
+const std::string WriteToFlowFileTestProcessor::OnTriggerLogStr = "WriteToFlowFileTestProcessor::onTrigger executed";
+const std::string WriteToFlowFileTestProcessor::OnUnScheduleLogStr = "WriteToFlowFileTestProcessor::onUnSchedule";
+
+core::Relationship WriteToFlowFileTestProcessor::Success("success", "success operational on the flow record");
+
+
+void WriteToFlowFileTestProcessor::initialize() {
+  setSupportedProperties({});
+  setSupportedRelationships({Success});
+}
+
+void WriteToFlowFileTestProcessor::onSchedule(core::ProcessContext*, core::ProcessSessionFactory*) {
+  logger_->log_info("%s", OnScheduleLogStr);
+}
+
+namespace {
+struct WriteBufferToFlowFile : public OutputStreamCallback {
+  const std::vector<uint8_t> buffer_;
+
+  explicit WriteBufferToFlowFile(const std::string& buffer) : buffer_(buffer.begin(), buffer.end()) {}
```

Comment:
There is no need to copy the content to the callback class. Just store a pointer to the original and use that buffer directly. (tip: `gsl::span`)